common/Scripts/helpers.sh

```diff
@@ -5,43 +5,42 @@
 # This function will only work if this script is sourced
 # by your bash shell.
 function gotosoluspkgs() {
-    cd $(dirname $(readlink "${BASH_SOURCE[0]}"))/../../
+    cd "$(dirname "$(readlink "${BASH_SOURCE[0]}")")/../../" || exit 1
 }
 
 # Goes to the root directory of the git repository
 function goroot() {
-    cd $(git rev-parse --show-toplevel)
+    cd "$(git rev-parse --show-toplevel)" || exit 1
 }
 
 # Push into a package directory
 function gotopkg() {
-    cd $(git rev-parse --show-toplevel)/packages/*/$1
+    cd "$(git rev-parse --show-toplevel)"/packages/*/"$1" || exit 1
 }
 
 # What provides a lib
 function whatprovides() {
-    grep $1 $(git rev-parse --show-toplevel)/packages/*/*/abi_libs
+    grep "$1" "$(git rev-parse --show-toplevel)"/packages/*/*/abi_libs
 }
 
 # What uses a lib
 function whatuses() {
-    grep $1 $(git rev-parse --show-toplevel)/packages/*/*/abi_used_libs
+    grep "$1" "$(git rev-parse --show-toplevel)"/packages/*/*/abi_used_libs
 }
 
 
 # Bash completions
 _gotopkg()
 {
 
-    _list=$(ls $(git rev-parse --show-toplevel)/packages/*/)
+    _list=$(ls "$(git rev-parse --show-toplevel)"/packages/*/)
 
-    local cur prev
+    local cur
     COMPREPLY=()
     cur=${COMP_WORDS[COMP_CWORD]}
-    prev="${COMP_WORDS[COMP_CWORD-1]}"
 
     if [[ $COMP_CWORD -eq 1 ]] ; then
-        COMPREPLY=( $(compgen -W "${_list}" -- ${cur}) )
+        mapfile -t COMPREPLY < <(compgen -W "${_list}" -- "${cur}")
```

Comment:
I think we should standardise on `readarray` instead as that is the newer (bash-4.2+) name for this?

Would probably also be nice to explain what the magic invocation is supposed to do here (what does `compgen` do for instance?)

Reply:
> I think we should standardise on readarray instead as that is the newer (bash-4.2+) name for this?

Sure, I just followed the shellcheck advice, which doesn't mention `readarray`.

> Would probably also be nice to explain what the magic invocation is supposed to do here (what does compgen do for instance?)

That would be nice, as I could also be sure that I haven't broken it :see_no_evil: